ls1intum/Artemis
# `Assessment`: Allow instructors to select csv export options before downloading course scores

### Checklist
#### General



#### Client


### Motivation and Context


When course scores get downloaded, the field separator as well as the quote strings are currently hardcoded and can not be customized by the user.


### Description

When the user clicks on the export button for downloading course scores, a new modal view is opened through which a custom field separator as well as custom quote strings can be selected.

Currently the newly added components are only used when exporting course scores, but in general they can be used everywhere in the project where csv exports are possible.

### Steps for Testing

Prerequisites:
- 1 Course
- 1 Student
- 1 Exercise, on which the student participated

1. Log in to Artemis
2. Navigate to Course Administration, select the course, then Scores 
3. Click the `Export as CSV` button
4. Select a field separator as well as a the quote string that should be used for generating the csv file
5. Click `Export` 
6. View the downloaded csv file in a text editor and make sure the previously selected options were applied 

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| csv-export-button.component.ts | 75% | 100% |
| csv-export-modal.component.ts | 85.71% | 96.42% |
| course-scores.component.ts | 73.84% | 94.72% |



### Screenshots


![csv-export](https://user-images.githubusercontent.com/9954674/160402405-b31fe394-41c5-459c-aff9-3154a8fc6fa1.gif)
